eos-phone-home

```diff
@@ -148,6 +153,43 @@ class PhoneHome(object):
 
         return None
 
+    def _get_metrics(self):
+        permissions_file = '/etc/metrics/eos-metrics-permissions.conf'
+        valid_environments = ['production', 'dev', 'test']
+
+        enabled = False
+        uploading_enabled = True
+        environment = 'production'
+
+        try:
+            config = configparser.ConfigParser()
+            config.read(permissions_file)
+            enabled = config['global'].getboolean('enabled', enabled)
+            uploading_enabled = config['global'].getboolean('uploading_enabled',
+                uploading_enabled)
+            environment = config['global'].get('environment', environment)
+        except Exception as e:
+            print("Unable to read %s, assuming fallback values." %
+                (permissions_file, ), e, file=sys.stderr)
+
+        # the event recorder replaces invalid environments with 'test'
+        if environment not in valid_environments:
```

Comment:
I don't mind dropping that provided we arrange that the server is similarly permissive. As @sgnn7 hinted above, the server is pedantic and defaults to erroring and hence discarding the whole event if invalid. If it's just a string on the server with no further validation, we can drop this too. FYI @rahmanShettia